prow/config/config_test.go

```diff
@@ -855,6 +897,12 @@ func TestLatestUsesImagePullPolicy(t *testing.T) {
 func checkKubekinsPresets(jobName string, spec *v1.PodSpec, labels, validLabels map[string]string) error {
 	service := true
 	ssh := true
+
+	if len(spec.InitContainers) > 0 && len(spec.Containers) > 0 {
+		// preset rules do not apply to complex pods
```

Comment:
/shrug

Reply:
hummmmmm - we need to map presets to each container I guess? @BenTheElder @spxtr

Reply:
Yeah presets make a lot less sense with a bunch of containers and initcontainers, but I don't expect users to put THOSE in the config in the future, just their test info. We can add presets more intelligently when Prow adds the initcontainers too, we will know which ones need what.

Reply:
> hummmmmm - we need to map presets to each container I guess? @BenTheElder @spxtr

Those would no longer be pod-presets at all, in which case we should possibly rethink them entirely :-)